run-tests.php

```diff
@@ -2158,14 +2141,14 @@ function run_test($php, $file, $env)
 		$restype[] = 'LEAK';
 	}
 
-	if ($warn) {
+	if ($warn_info !== null) {
 		$restype[] = 'WARN';
 	}
 
 	if (!$passed) {
 		if (isset($section_text['XFAIL'])) {
 			$restype[] = 'XFAIL';
-			$info = '  XFAIL REASON: ' . rtrim($section_text['XFAIL']);
+			$warn_info = '  XFAIL REASON: ' . rtrim($section_text['XFAIL']);
```

Comment:
> then it will be detected under the WARN restype

@nikic Where is it detected? I've changed to `$warn_info` and haven't noticed any change.

Reply:
I think the logic is right, since the check happens on L2144. but:

1. `info` is a general term, but `warn info` indicates it should be warning and warning only.
2. if other people want to check if there is warning or not below this line, there is no straightforward way to do so. There is no more `$warn`, and `$warn_info !== null` cannot be used neither. 

I guess this is what nikic means 
> detected under the WARN restype

Reply:
@jhdxr Have reverted these changes than :)